packages/bpk-component-image/src/withLoadingBehavior.js

```diff
@@ -0,0 +1,50 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { wrapDisplayName } from 'bpk-react-utils';
+
+export default function withLoadingBehavior(Component) {
+  class WithLoadingBehavior extends React.Component {
+    constructor() {
+      super();
+
+      this.onLoaded = this.onLoaded.bind(this);
+
+      this.state = {
+        loading: true,
+      };
+    }
+
+    onLoaded() {
+      this.setState({ loading: false });
+    }
+
+    render() {
+      return (
+        <Component
+          onLoaded={this.onLoaded}
```

Comment:
I think `onLoaded` should be renamed `onLoad` if there's not a collision with a React provided callback